node/src/components/consensus/highway_core/state.rs

```diff
@@ -551,6 +558,17 @@ impl<C: Context> State<C> {
                 }
             }
         }
+        // All endorsed units from the panorama of this wunit.
+        let endorsements_in_panorama = panorama
+            .iter_correct_hashes()
+            .flat_map(|hash| self.unit(hash).claims_endorsed())
+            .collect::<HashSet<_>>();
+        if endorsements_in_panorama
+            .iter()
+            .any(|&e| !wunit.endorsed.iter().any(|h| h == e))
```

Comment:
As per discussion with @afck , `wunit.endorsed` should be a `BTreeSet`

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/527/commits/475c972cc957668e1dcd56c40dee30d0246f1a43